functional_annotation/functional_annotation.md

```diff
@@ -0,0 +1,239 @@
+
+As with most things in bioinformatics, there are many ways to annotate amino acid sequences. 
+We will be using kofamscan, a tool that uses hidden markov models built from KEGG orthologs to assign KEGG ortholog numbers to new amino acid sequences. 
+Hidden markov models work well for protein annotation because they weight the importance of each amino acid in determining the final assignment. 
+Look at the sequence logo figure below.
+
+![](https://i.imgur.com/5Cq8Tiy.png)
+
+This figure is a logo depicting the PFAM HMM for rpsG. 
+rpsG encodes 30S ribosomal protein S7, and it is a highly conserved protein. 
+The HMM was built from hundreds of rpsG protein sequences. 
+At each position of the protein, the logo depicts the liklihood of seeing a specific amino acid. 
+The larger the amino acid is in the logo, the more likely it is to be observed at that position. 
+In positions where no amino acid is visible, it is less important which amino acid occurs there. 
+This encoding is more flexible than something like Hamming distance or BLAST because it incorporates biological importance of amino acid positionality. 
+This approach works well on novel amino acid sequences that are not closely related to anything currently housed in databases.
+
+`kofamscan` is a tool released by the KEGG that includes HMMs built from each KEGG ortholog. 
+Using `kofamscan`, we can assign KEGG orthologs to our amino acid sequences. 
+This allows us to take advantage of KEGG pathway information.
+
+`kofamscan` is run by using the `exec_annotation` command:
+
+```
+exec_annotation --help
+```
+
+`kofamscan` needs a few database files in order to run.
+Let's download them.
+
+```
+wget ftp://ftp.genome.jp/pub/db/kofam/ko_list.gz      # download the ko list 
+wget ftp://ftp.genome.jp/pub/db/kofam/profiles.tar.gz # download the hmm profiles
+```
+
+The first file is the KEGG orthlog (KO) list.
+The second file is the HMM profiles for the KOs.
+
+We also have to unzip and untar the relevant files:
+```
+gunzip ko_list.gz
+tar xf profiles.tar.gz
+```
+
+kofamscan runs using a config file. 
+Using nano, build a config file that looks like this:
+
+```
+nano config.yaml
+```
+
+```
+# Path to your KO-HMM database
+# A database can be a .hmm file, a .hal file or a directory in which
+# .hmm files are. Omit the extension if it is .hal or .hmm file
+profile: ./profiles
+
+# Path to the KO list file
+ko_list: ko_list
+
+# Path to an executable file of hmmsearch
+# You do not have to set this if it is in your $PATH
+# hmmsearch: /usr/local/bin/hmmsearch
+
+# Path to an executable file of GNU parallel
+# You do not have to set this if it is in your $PATH
+# parallel: /usr/local/bin/parallel
+
+# Number of hmmsearch processes to be run parallelly
+cpu: 8
+```
+
```

Comment:
```suggestion

> You might notice the `$PATH` variable above. We didn't get to cover what this special variable is in a Unix-like environment, but it can be useful to know about. The PATH variable holds all the locations (directories) our command line will automatically search when looking for a program we try to run. If interested in more about this and how to modify our PATH, you can find more about it [here](https://astrobiomike.github.io/unix/modifying_your_path) 👍 

```

Reply:
```suggestion
You might notice the `$PATH` variable above. We didn't get to cover what this special variable is in a Unix-like environment, but it can be useful to know about. The PATH variable holds all the locations (directories) our command line will automatically search when looking for a program we try to run. If interested in more about this and how to modify our PATH, you can find more about it [here](https://cac-staff.github.io/swc-bash-lesson/j05-hpc.html) 
```